src/parsers/perf/perfparser.cpp

```diff
@@ -1523,6 +1524,52 @@ void PerfParser::startParseFile(const QString& path)
                 &QEventLoop::quit);
         loop.exec();
     });
+
+    if (diffFile.isEmpty()) {
+        return;
+    }
+
+    PerfParser otherData;
+
+    otherData.startParseFile(diffFile);
+    QEventLoop loop;
+    connect(&otherData, &PerfParser::parsingFinished, &loop, &QEventLoop::quit);
+    connect(&otherData, &PerfParser::parsingFailed, &loop, &QEventLoop::quit);
+    connect(&otherData, &PerfParser::parsingFailed, this, [this](const QString& error) { emit parsingFailed(error); });
+
+    connect(&otherData, &PerfParser::bottomUpDataAvailable, &otherData,
+            [this, &otherData](const Data::BottomUpResults& otherResults) {
+                if (m_bottomUpResults.root.children.size() > 0) {
+                    this->m_callerCalleeResults = {};
+
+                    const auto diffResults =
+                        Data::BottomUpResults::mergeBottomUpResults(m_bottomUpResults, otherResults);
```

Comment:
this should diff, not merge, or? the next patch in the series changes this, but it should be part of this patch already